server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,26 +342,137 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
     }
   }
 
-  private void loadSnapshotAndInitStateRef()
+  private void loadAllLookupsAndInitStateRef()
   {
-    if (lookupSnapshotTaker != null) {
-      ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
+    List<LookupBean> lookupBeanList = new ArrayList<>();
+    if (lookupConfig.getEnableLookupSyncOnStartup()) {
+      String tier = lookupListeningAnnouncerConfig.getLookupTier();
+      lookupBeanList = getLookupListFromCoordinator(tier);
+      if (lookupBeanList == null) {
+        LOG.info("Coordinator is unavailable. Loading saved snapshot instead");
+        lookupBeanList = getLookupListFromSnapshot();
+      }
+    } else {
+      lookupBeanList = getLookupListFromSnapshot();
+    }
+      if (lookupBeanList != null) {
+        ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
+        ListeningScheduledExecutorService executorService = MoreExecutors.listeningDecorator(
```

Comment:
This is done a number of places, can you file an issue please to get a `ListeningScheduledExecutorService` method in `Execs`?